pkg/apis/extensions/v1alpha1/types_operatingsystemconfig.go

```diff
@@ -227,12 +227,25 @@ const (
 	OperatingSystemConfigSecretDataKey = "cloud_config" // #nosec G101 -- No credential.
 )
 
+// CgroupDriverName is a string denoting the CRI cgroup driver.
+type CgroupDriverName string
+
+const (
+	// CgroupDriverCgroupfs sets the cgroup driver to cgroupfs.
+	CgroupDriverCgroupfs CgroupDriverName = "cgroupfs"
+	// CgroupDriverSystemd sets the cgroup driver to system.
+	CgroupDriverSystemd CgroupDriverName = "systemd"
+)
+
 // CRIConfig contains configurations of the CRI library.
 type CRIConfig struct {
 	// Name is a mandatory string containing the name of the CRI library. Supported values are `containerd`.
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
 	// +kubebuilder:validation:Enum="containerd"
 	Name CRIName `json:"name"`
+	// CgroupDriver configures the CRI's cgroup driver. Supported values are `cgroupfs` or `systemd`.
+	// +optional
+	CgroupDriver *CgroupDriverName `json:"cgroupDriver,omitempty"`
```

Comment:
Shouldn't this field be immutable once set? I see that the gardenlinux extension mutates the corresponding containerd field only when there are no running containerd tasks (I assume this means containers?) - see https://github.com/gardener/gardener-extension-os-gardenlinux/blob/e8d59651b76c2bf8b132a16a3a1876f05cbaffaa/pkg/gardenlinux/scripts/containerd_cgroup_driver.sh#L26-L35. I am not sure whether it is safe to change the cgroup driver when there are already running containers or containerd tasks.

cc @MrBatschner

Reply:
In most cases the cgroup driver should not change during runtime and making it immutable is reasonable.

OTHO, it should be allowed to change `cgroupDriver = cgroupfs` -> `cgroupDriver = null`, i.e. to make it unmanaged again in case some other mechanism takes over management. It depends very much on the use-case, OS and probably container runtime. In the end, the `OSC` is an internal/expert API and I think it's fine to leave it mutable.